content/posts/cicd_ha.md

```diff
@@ -0,0 +1,132 @@
+---
+authors:
+  - "David Cañadillas"
+title: "CI/CD High Availability: Misunderstanding the Concept"
+date: 2019-10-28T01:51:10+01:00
+showDate: true
+draft: false
+tags: ["jenkins","jenkins x", "CI/CD pipelines","High Availability"]
+---
+
+High availability scenarios are key configurations for all mission critical software platforms and solutions at every company.
+
+But in my experience from every technology conversation, I've found out that High Availability (HA from now on) is one of those terms that is usually misunderstood when talking about CI/CD. The reason? Because HA can be a subjective topic depending on who you are talking to (IT, Development, Operations, end users, etc.).
+
+I am writing this post to clear things out about the "myth" of HA in CI/CD and what it means from a solution architecture point of view. So, don't expect a post with a lot of code or configuration examples. It is about architecture concepts, business needs and its technical alignment to help finding out the best practices for *"highly available"* CI/CD pipelines. 
+
+
+## What is HA and why is it relative?
+
+Even though [the definition at Wikipedia](https://en.wikipedia.org/wiki/High_availability) is not a true "official" source, I like how it defines HA:
+
+> *High availability (HA) is a characteristic of a system, which aims to ensure an agreed level of operational performance, usually uptime, for a higher than normal period.*
+
+The words *agreed level of operational performance* show some very important topics to clarify:
+
+* *Operational performance* is defined by the use case and the business. Not from the technical components or the software being used.
+* The software architecture and its technical definition depends on the *agreed level* of that operational performance.
+
+Let's use an example. If we think about the service that [Amazon retail](https://amazon.com) users are expecting, any purchase transaction using the web portal cannot be interrupted for even a few seconds. So, web services, front end applications, databases, middleware, payment channels  and other services or microservices, etc. cannot interrupt or block that user transaction. That means that every server needs to be up and running all the time, or at least to recover at the same point in a sub-second basis. 
+
+This is just because the business requires that any user transaction and interaction with the application cannot be interrupted at all, considering that an interruption is something blocked during 3 seconds or more (users don't like to wait more than 3 seconds to refresh a web page to the next step). And if that doesn't happen, I - as a user - will be doing the transaction from a different web retailer or competitor. The business then requires a very restrictive HA scenario.
```

Comment:
I might change to "an interruption is something blocked *for* 3 seconds or more"